src/planwise/component/engine.clj

```diff
@@ -52,6 +52,36 @@
          (sort-by :capacity)
          reverse)))
 
+(defn- compute-provider
+  [props provider]
+  (let [{:keys [id provider-id action raster capacity]} provider
+        {:keys [update? demand-raster project-capacity project-id provider-set-id]} props
+        path (if action
+               (str "data/scenarios/" project-id "/coverage-cache/" provider-id ".tif")
+               (str "data/coverage/" provider-set-id "/" raster ".tif"))
+        coverage-raster (raster/read-raster path)
+        scaled-capacity (* capacity project-capacity)
+        population-reachable (demand/count-population-under-coverage demand-raster coverage-raster)]
+
+    (if update?
+      {:unsatisfied population-reachable}
```

Comment:
Why not return the updated provider here? Eg. `(assoc provider :unsatisfied population-reachable)`.